### PR TITLE
CONTRIBUTING.md - DRAFT

### PR DESCRIPTION
**Pull request by theapache64**
## What is this?

This is an early version of `CONTRIBUTING.md` to help other fellow compose enthusiasts to understand the opportunities and how to start contributing to `compose-jb`.

## Original Source

https://kotlinlang.slack.com/archives/C01D6HTPATV/p1616075631036600?thread_ts=1616048882.029100&cid=C01D6HTPATV 

Feel free to edit as you see fit.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# Contributing 🙌
```
Comment (olonho) [inline CONTRIBUTING.md:1]: Having emojis is not the style we follow in other docs.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+> When contributing to this repository, please first discuss the change you wish to make via issue,
+email, or any other method with the owners of this repository before making a change. 
+
+So you want to be a part of "The Compose Revolution" and that's great. Here are some contribution opportunities, in order of difficulty.
```
Comment (olonho) [inline CONTRIBUTING.md:6]: It's somewhat emotional for an official document.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+**Difficulty** : Beginner
+
+Go through the docs and tutorials, following the steps in each one. Identify mistakes where things are out of date or don't align with the code or are confusing and submit fixes.
+
+
+## Community Contributions 👏
+
+**Difficulty** : Beginner
+
+Building cool stuff and posting it on twitter and reddit can help inspire other people to join the community and build cool stuff. You use this as an opportunity to learn Compose, find bugs, and occasionally fix bugs too, all at the same time.  And if you build cool things and share them in open source such that people can share/reuse your code, that can help them get started. Getting people excited and building a great community is important for the success of Compose for Desktop.
```
Comment (olonho) [inline CONTRIBUTING.md:19]: We probably need to be more concrete in this document on technical side, and less on social media activity.

File: CONTRIBUTING.md (hunk 1 above)
Comment (theapache64) [reply CONTRIBUTING.md:1]: Understood. Will remove.

File: CONTRIBUTING.md (hunk 2 above)
Comment (theapache64) [reply CONTRIBUTING.md:6]: Okay. What do you suggest?

File: CONTRIBUTING.md (hunk 3 above)
Comment (theapache64) [reply CONTRIBUTING.md:19]: Okay. What things do you want to include?

File: CONTRIBUTING.md (hunk 1 above)
Comment (theapache64) [reply CONTRIBUTING.md]: Resolved via https://github.com/JetBrains/compose-jb/pull/511/commits/d64043f4cc247009deb3e6e8f7c35a8f7bfba622

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
+# Contributing
+
+> When contributing to this repository, please first discuss the change you wish to make via issue,
```
Comment (B1ggDave) [inline CONTRIBUTING.md:3]: > When contributing to this repository, please first discuss the change you wish to make with the owners of this repository before making any changes. You can contact the owners by email or via an issue.

or if it is truly about any means of communication then this is enough --

> When contributing to this repository, please first discuss the change you wish to make with the owners of this repository before making any changes.

File: CONTRIBUTING.md (hunk 2 above)
Comment (B1ggDave) [reply CONTRIBUTING.md:6]: Are you looking for ways to contribute to Compose? Here are some opportunities you can explore, in order of difficulty.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+email, or any other method with the owners of this repository before making a change. 
+
+So you want to be a part of "The Compose Revolution" and that's great. Here are some contribution opportunities, in order of difficulty.
+
+## Documentation & Tutorials
+
+**Difficulty** : Beginner
+
+Go through the docs and tutorials, following the steps in each one. Identify mistakes where things are out of date or don't align with the code or are confusing and submit fixes.
```
Comment (B1ggDave) [inline CONTRIBUTING.md:12]: Go through the docs and tutorials, and follow the steps. Identify mistakes and parts that are confusing, and submit fixes.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+
+## Community Contributions
+
+**Difficulty** : Beginner
+
+Building cool stuff and posting it on twitter and reddit can help inspire other people to join the community and build cool stuff. You use this as an opportunity to learn Compose, find bugs, and occasionally fix bugs too, all at the same time.  And if you build cool things and share them in open source such that people can share/reuse your code, that can help them get started. Getting people excited and building a great community is important for the success of Compose for Desktop.
```
Comment (B1ggDave) [inline CONTRIBUTING.md:19]: Inspire other people to try Compose by building cool stuff and posting it on Twitter and Reddit. Sharing your code as open source can also help people get started. Getting people excited and building a great community is important for the success of Compose for Desktop.

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+
+## Samples
+
+**Difficulty** : Intermediate
+
+ Often the samples will be slightly out of date (using an older release of Compose for Desktop) or will have subtle bugs. Cleaning up such bugs helps make the samples look good.
```
Comment (B1ggDave) [inline CONTRIBUTING.md:25]: The samples themselves may be slightly out of date (using an older release of Compose for Desktop) or could have subtle bugs. Cleaning up these bugs helps keep the samples looking good.

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+
+
+## Widgets
+
+**Difficulty** : Expert
+
+Widgets are usually relatively well-scoped and require less domain knowledge than core contributions.  For example, [CEF](https://github.com/JetBrains/compose-jb/tree/master/cef) doesn't currently work on mac osx.  If you have a mac and are willing to dive into the technicals, that would be very helpful thing to solve.
```
Comment (B1ggDave) [inline CONTRIBUTING.md:32]: Widgets are relatively well-scoped and require less domain knowledge than core contributions.  For example, [CEF](https://github.com/JetBrains/compose-jb/tree/master/cef) doesn't currently work on macOS.  If you have a Mac and are willing to dive into this, that would be very helpful.

alt:
Widgets are relatively well-scoped and require less domain knowledge than core contributions. Helping to fix issues in these widgets would be incredibly helpful.

-- I don't know if you should put an example as this immediately makes this document outdated, as soon as the issue is fixed.

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+
+## Bugs
+
+**Difficulty** : Depends on the bug
```
Comment (B1ggDave) [inline CONTRIBUTING.md:36]: Varies

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+
+You'll need to use your judgement to find ones that look interesting and are also small enough to tackle.  This requires a bit more knowledge of Compose and would be especially hard for someone who doesn't have lots of experience contributing to open source, so I'd work your way up to this point, aiming for some of the earlier bullet points first.
```
Comment (B1ggDave) [inline CONTRIBUTING.md:38]: You'll need to use your judgment to find bugs that are within your capabilities to tackle. This requires a bit more knowledge of Compose and would be especially hard for someone who doesn't have lots of experience contributing to open source. You should work your way up to this by contributing in other ways.

File: CONTRIBUTING.md (hunk 4 above)
Comment (theapache64) [reply CONTRIBUTING.md:3]: Latter sounds good. Will update.

File: CONTRIBUTING.md (hunk 2 above)
Comment (theapache64) [reply CONTRIBUTING.md:6]: That sounds better. Will update.

File: CONTRIBUTING.md (hunk 5 above)
Comment (theapache64) [reply CONTRIBUTING.md:12]: That's better. Will update.

File: CONTRIBUTING.md (hunk 6 above)
Comment (theapache64) [reply CONTRIBUTING.md:19]: Cool. that's less and conveys the same message. Will update

File: CONTRIBUTING.md (hunk 7 above)
Comment (theapache64) [reply CONTRIBUTING.md:25]: Cool. Will update

File: CONTRIBUTING.md (hunk 8 above)
Comment (theapache64) [reply CONTRIBUTING.md:32]: Yeah. that makes sense. Will omit the example.

File: CONTRIBUTING.md (hunk 9 above)
Comment (theapache64) [reply CONTRIBUTING.md:36]: Okay. Will update to `Varies by bug`. :+1:

File: CONTRIBUTING.md (hunk 10 above)
Comment (theapache64) [reply CONTRIBUTING.md:38]: Will update. :+1:

File: CONTRIBUTING.md (hunk 2 above)
Comment (theapache64) [reply CONTRIBUTING.md]: Updated via https://github.com/JetBrains/compose-jb/pull/511/commits/59f6636402682a1f8825ccfc0a74c2bc0b8c3813

File: CONTRIBUTING.md (hunk 4 above)
Comment (theapache64) [reply CONTRIBUTING.md]: @B1ggDave  Updated via https://github.com/JetBrains/compose-jb/pull/511/commits/59f6636402682a1f8825ccfc0a74c2bc0b8c3813

File: CONTRIBUTING.md (hunk 5 above)
Comment (theapache64) [reply CONTRIBUTING.md]: @B1ggDave  Updated via https://github.com/JetBrains/compose-jb/pull/511/commits/59f6636402682a1f8825ccfc0a74c2bc0b8c3813

File: CONTRIBUTING.md (hunk 6 above)
Comment (theapache64) [reply CONTRIBUTING.md]: @B1ggDave  Updated via https://github.com/JetBrains/compose-jb/pull/511/commits/59f6636402682a1f8825ccfc0a74c2bc0b8c3813

File: CONTRIBUTING.md (hunk 7 above)
Comment (theapache64) [reply CONTRIBUTING.md]: @B1ggDave  Updated via https://github.com/JetBrains/compose-jb/pull/511/commits/59f6636402682a1f8825ccfc0a74c2bc0b8c3813

File: CONTRIBUTING.md (hunk 8 above)
Comment (theapache64) [reply CONTRIBUTING.md]: @B1ggDave  Updated via https://github.com/JetBrains/compose-jb/pull/511/commits/59f6636402682a1f8825ccfc0a74c2bc0b8c3813

File: CONTRIBUTING.md (hunk 9 above)
Comment (theapache64) [reply CONTRIBUTING.md]: @B1ggDave  Updated via https://github.com/JetBrains/compose-jb/pull/511/commits/59f6636402682a1f8825ccfc0a74c2bc0b8c3813

File: CONTRIBUTING.md (hunk 10 above)
Comment (theapache64) [reply CONTRIBUTING.md]: @B1ggDave  Updated via https://github.com/JetBrains/compose-jb/pull/511/commits/59f6636402682a1f8825ccfc0a74c2bc0b8c3813

**Comment by adamglin0** [comment]
Why has this PR come to a standstill? I believe many people are interested in Compose, but they don't know how to make contributions.

**Pull request closed without merging**